src/util/types.ts

```diff
@@ -1079,6 +1084,8 @@ export type AxisDomain =
  */
 export type NumberDomain = [min: number, max: number];
 
+export type CategoricalDomain = (number | string | Date)[];
```

Comment:
I assume more specifically: 

```suggestion
type CategoricalDomainValue = number | string | Date
export type CategoricalDomain = [CategoricalDomainValue, CategoricalDomainValue];
```

This would capture the fact that it must be exactly two values, right?

Reply:
I am not convinced it's always two values. If an axis is `type = category` then its domain should be a list of all categories, right? When given `['fruits', 'vegetables', 'dairy', 'meat']`, recharts will keep all the categories. Unlike numerical axis of course.

The code certainly allows multiple values.

Reply:
Right, I totally forgot about the PointScale, which simply has values in the domain. Sorry.